thirdeye-server/src/main/java/ai/startree/thirdeye/auth/AuthorizationManager.java

```diff
@@ -58,6 +62,12 @@ public class AuthorizationManager {
   private final EnumerationItemManager enumerationItemManager;
   private final AnomalyManager anomalyManager;
 
+  // cache resource id's to their respective auth namespaces
+  private final Cache<Long, String> namespaceCache = CacheBuilder.newBuilder()
```

Comment:
Did you check out `LoadingCache`? It's probably a subclass of this.

Reply:
Ah ya I can use that. The difference is in the `get` method: Cache.get needs the loading function, whereas LoadingCache.get uses loading function from builder.

Reply:
Ended up using regular cahce, since the loader part depends on the entity type (alert loaded by alert manager, anomaly loaded by anomaly manager)